From 88d4e5d79ca5ded0f9e88af67044d53342ff0897 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 22 Oct 2009 17:39:27 -0400
Subject: [PATCH 1/2] Force cell renderer to be a specific height

Before it was growing and shrinking all the time causing,
flicker.
---
 gui/simple-greeter/gdm-cell-renderer-timer.c |   12 ++----------
 1 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/gui/simple-greeter/gdm-cell-renderer-timer.c b/gui/simple-greeter/gdm-cell-renderer-timer.c
index 7918b48..9453d7e 100644
--- a/gui/simple-greeter/gdm-cell-renderer-timer.c
+++ b/gui/simple-greeter/gdm-cell-renderer-timer.c
@@ -108,19 +108,11 @@ gdm_cell_renderer_timer_get_size (GtkCellRenderer *cell,
         }
 
         if (width != NULL) {
-                if (cell->width >= 0) {
-                        *width = cell->width;
-                } else {
-                        *width = renderer->priv->value <= G_MINDOUBLE?  0 : 64;
-                }
+                *width = cell->xpad * 2 + 42;
         }
 
         if (height != NULL) {
-                if (cell->height >= 0) {
-                        *height = cell->height;
-                } else {
-                        *height = renderer->priv->value <= G_MINDOUBLE?  0 : 64;
-                }
+                *height = cell->ypad * 2 + 42;
         }
 }
 
-- 
1.6.5.rc2


From a87b05fac2c2de7cff80a183e397f33416bfc893 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 22 Oct 2009 17:45:13 -0400
Subject: [PATCH 2/2] Drop set_operator (SOURCE) in cell renderer

I'm not sure why I was doing it, when it looks
find using OVER, so I'm going to take it out.
---
 gui/simple-greeter/gdm-cell-renderer-timer.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/gui/simple-greeter/gdm-cell-renderer-timer.c b/gui/simple-greeter/gdm-cell-renderer-timer.c
index 9453d7e..0fb404c 100644
--- a/gui/simple-greeter/gdm-cell-renderer-timer.c
+++ b/gui/simple-greeter/gdm-cell-renderer-timer.c
@@ -164,7 +164,6 @@ draw_timer (GdmCellRendererTimer *renderer,
         cairo_arc (context, 0, 0, radius + 1, 0, 2 * G_PI);
         cairo_fill (context);
 
-        cairo_set_operator (context, CAIRO_OPERATOR_SOURCE);
         cairo_set_source_rgb (context,
                               bg->red / 65535.0,
                               bg->green / 65535.0,
-- 
1.6.5.rc2
